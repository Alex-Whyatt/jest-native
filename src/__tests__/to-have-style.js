import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import { render } from 'native-testing-library';

describe('.toHaveStyle', () => {
  test('handles positive test cases', () => {
    const styles = StyleSheet.create({ container: { color: 'white' } });
    const { getByTestId } = render(
      <View
        testID="container"
        style={[{ backgroundColor: 'blue', height: '100%' }, styles.container]}
      >
        <Text>Hello World</Text>
      </View>,
    );

    const container = getByTestId('container');

    expect(container).toHaveStyle({ backgroundColor: 'blue', height: '100%' });
    expect(container).toHaveStyle([{ backgroundColor: 'blue' }, { height: '100%' }]);
    expect(container).toHaveStyle({ backgroundColor: 'blue' });
    expect(container).toHaveStyle({ height: '100%' });
    expect(container).toHaveStyle({ color: 'white' });
  });

  test('handles negative test cases', () => {
    const { getByTestId } = render(
      <View testID="container" style={{ backgroundColor: 'blue', color: 'black', height: '100%' }}>
        <Text>Hello World</Text>
      </View>,
    );

    const container = getByTestId('container');

    expect(() => expect(container).toHaveStyle({ fontWeight: 'bold' })).toThrowError();
    expect(() => expect(container).not.toHaveStyle({ color: 'black' })).toThrowError();
  });
});
