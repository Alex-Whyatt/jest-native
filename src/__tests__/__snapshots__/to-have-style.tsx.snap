// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.toHaveStyle handles negative test cases 1`] = `
"expect(element).toHaveStyle()

- Expected
+ Received

  backgroundColor: blue;
  transform: [
    {
-     "scale": 1
+     "scale": 2
+   },
+   {
+     "rotate": "45deg"
    }
  ];"
`;

exports[`.toHaveStyle handles transform when transform undefined 1`] = `
"expect(element).toHaveStyle()

- Expected
+ Received

- transform: [
-   {
-     "scale": 1
-   }
- ];
+ transform: undefined;"
`;
